### PR TITLE
issue 80: Allow a message to be sent to update contract config

### PR DESCRIPTION
**Pull request by wng-figure**
issue 80: Allow a message to be sent to update contract config. 

this version has multiple commits squashed.

draft version comments are here: https://github.com/wng-figure/ats-smart-contract/pull/1

### CODE REVIEW

**Comment by leeduan** [comment]
@ktalley-figure mind providing some feedback cycles here?

File: src/contract_info.rs [hunk 1]
```diff
@@ -252,6 +252,87 @@ pub fn migrate_contract_info(
...
+            contract_info.bid_required_attributes = bid_required_attributes.clone();
+        }
+    }
+
+    set_contract_info(store, &contract_info)?;
+
+    get_contract_info(store)
+}
+
+pub fn modify_contract_info(
```
Comment (leeduan) [inline src/contract_info.rs:290]: this is essentially a copy of migrate_contract_info right?

wondering for the repeated pieces if we should move them to another fun 🤷

File: src/contract.rs [hunk 2]
```diff
@@ -1106,6 +1131,92 @@ fn reverse_bid(
...
+        .map(|kv_ask: StdResult<Pair<AskOrderV1>>| {
+            let (_, ask_order) = kv_ask.unwrap();
+            ask_order
+        })
+        .collect();
+    if !ask_orders.is_empty() {
+        match &msg.ask_required_attributes {
+            None => (),
+            Some(_) => {
+                return Err(ContractError::InvalidFields {
```
Comment (leeduan) [inline src/contract.rs:1160]: wondering if these errors should be tailored to why?

File: src/contract.rs [hunk 3]
```diff
@@ -1106,6 +1131,92 @@ fn reverse_bid(
...
+                }
+            }
+        }
+    }
+
+    modify_contract_info(deps, &msg)?;
+
+    let response = Response::new();
+
+    Ok(response)
```
Comment (leeduan) [inline src/contract.rs:1217]: this is so much simpler; nice

**Review by leeduan — APPROVED** [review]

**Comment by wng-figure** [comment]
> 

bumped version

**Review by ktalley-figure — APPROVED** [review]

**Pull request merged**